k9mail-library/src/test/java/com/fsck/k9/mail/store/imap/ImapResponseHelper.java

```diff
@@ -43,4 +50,49 @@ public static ImapResponse createImapResponse(String response) throws IOExceptio
         }
         return ids;
     }
+
+    static List<ImapResponse> createFolderOpenedResponse(int openMode, Long uidValidity, Long highestModSeq,
```

Comment:
I see the intention here to make the code shorter, but this is too much logic in a test, which obscures the actual test cases and makes them less isolated.

If we have some imap response in a test that works, the cases where this stops working should be very very rare - why would a reply that is valid in a test no longer be valid at any point? This method is far from trivial, and changing what it does also changes the meaning of all unit tests it is called in, in ways that are very hard to predict.